Unix/tests/wsman/test_wsmanparser.cpp

```diff
@@ -1487,6 +1487,138 @@ NitsTestWithSetup(TestFaultResponse2, TestParserSetup)
+        PAL_T("xmlns:p=\"http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/__ExtendedStatus\" ")
+        PAL_T("xmlns:cim=\"http://schemas.dmtf.org/wbem/wscim/1/common\" ")
+        PAL_T("xsi:type=\"p:__ExtendedStatus_Type\">")
+        PAL_T("<p:Description xsi:nil=\"true\" xsi:type=\"cim:cimString\"/>")
+        PAL_T("<p:Operation xsi:type=\"cim:cimString\">GetObject</p:Operation>")
+        PAL_T("<p:ParameterInfo xsi:type=\"cim:cimString\">Win32_Process.Handle=&quot;29&quot;</p:ParameterInfo>")
+        PAL_T("<p:ProviderName xsi:type=\"cim:cimString\">CIMWin32</p:ProviderName><p:StatusCode xsi:nil=\"true\" xsi:type=\"cim:cimUnsignedInt\"/>")
+        PAL_T("</p:__ExtendedStatus></f:ExtendedError></f:ProviderFault>")
+        PAL_T("</f:Message></f:WSManFault></s:Detail></s:Fault></s:Body></s:Envelope>");
+
+    memset(&wsheaders, 0, sizeof(wsheaders));
+
+    XML_Init(&xml);
+
+    XML_RegisterNameSpace(&xml, 's',
+                          ZT("http://www.w3.org/2003/05/soap-envelope"));
+    
+    XML_RegisterNameSpace(&xml, 'a',
+                          ZT("http://schemas.xmlsoap.org/ws/2004/08/addressing"));
+
+    XML_RegisterNameSpace(&xml, 'w',
+                          ZT("http://schemas.xmlsoap.org/ws/2005/06/management"));
+
+    XML_RegisterNameSpace(&xml, 'b',
+                          ZT("http://schemas.dmtf.org/wbem/wsman/1/cimbinding.xsd"));
+
+    XML_SetText(&xml, data);
+
+
+    if (!NitsCompare(MI_RESULT_OK, WS_ParseSoapEnvelope(&xml), PAL_T("Parse soap envelope error")))
+    {
+        goto cleanup;
+    }
+
+    if (!NitsCompare(MI_RESULT_OK, WS_ParseWSHeader(&xml, &wsheaders, USERAGENT_UNKNOWN), PAL_T("Parse header error")))
+    {
+        goto cleanup;
+    }
+
+    if (!NitsCompare(MI_TRUE, wsheaders.foundAction, PAL_T("No action in header")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompare(wsheaders.rqtAction, WSMANTAG_ACTION_FAULT_ADDRESSING, PAL_T("Action is not Fault ADDRESSING Response")))
+    {
+        goto cleanup;
+    }
+
+    if (!NitsCompare(MI_RESULT_OK, WS_ParseFaultBody(&xml, &fault), PAL_T("Unable to retrieve fault")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.code, PAL_T("http://www.w3.org/2003/05/soap-envelope:Sender"), 
+                           PAL_T("Mismatch of fault code")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.subcode, 
+                           PAL_T("http://schemas.xmlsoap.org/ws/2004/08/addressing:DestinationUnreachable"), 
+                           PAL_T("Mismatch of fault subcode")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.reason, PAL_T("The WS-Management service cannot process the request. The service cannot find the resource identified by the resource URI and selectors. "), PAL_T("Mismatch of fault reason")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.detail, PAL_T("CIMWin32"), 
```

Comment:
This does not seem right, especially as there is nothing useful in the details. Understandably the extended status->description is empty, which is not helpful, but maybe this string should default to the fault.reason if that is the case.